src/machines/commandBarMachine.ts

```diff
@@ -478,7 +475,7 @@ export const commandBarMachine = setup({
         {
           target: 'Closed',
           guard: 'Command has no arguments',
-          actions: ['Execute command'],
+          actions: ['Execute command', 'Clear selected command'],
```

Comment:
Remember that actions are "fire and forget". If anything is expecting selected command to be cleared, we are racing.

Reply:
Ah great point, this code predates that realization! Let me rewire it real quick to have an actor on an intermediate state

Reply:
Ah actually, I'm not sure I see the race potential here, since these actions are assignments internal to the machine itself. I _do_ see a race potential with the "Execute command" **action**, that should be an actor on an intermediate state for sure.

Does that sound right to you? I don't think actions that are solely `assign` actions are not worrisome because they are synchronous and internal to the state machine.

Reply:
Because the state transition and the context assignment are then guaranteed to be always in sync, there is no async assignment or action occurring

Reply:
We'd be relying on the particular behavior of XState v5 by assuming the actions happen in a sequential, determined order.

Otherwise I believe you're right that the actions, not invoking any kind of concurrency themselves, will be deterministic and doesn't matter if they are updating the state machine, etc. Just note we're heavily relying on how XState v5 handles actions.

Reply:
Remember too the actions don't need to finish before moving on

Reply:
Yeah totally, but sequential actions is canonical XState behavior from v5 onward, and in v4 it had `predictableActionOrder` or something. I feel comfortable relying on that mechanic for the future.

I think actions do need to finish before XState moves on its lifecycle, as long as they are synchronous. But you're right that we should confirm that behavior. @farskid do you know if that is always true? If it isn't that would make even assignment actions untrustworthy I think

Reply:
I think this is useful discussion but I'll take you up on the offer to move forward and observe if there are issues with this approach in the future. I feel pretty confident in it

Reply:
Retracting: XState treats "assign actions" _as their own special action_ :facepalm: https://stately.ai/docs/actions#assign-action Kill me. You're right Frank, all good here